test/select.spec.js

```diff
@@ -636,6 +639,31 @@ describe('ui-select tests', function () {
     expect(el.find('.select2-search-choice-close').length).toEqual(1);
   });
 
+  it('should clear selection (with object as source)', function() {
```

Comment:
Let's get this landed and open another issue for doing some test cleanup? Happy to take a stab at it in a few days (probably over the weekend)!